### PR TITLE
Isolate calico files

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

**Comment by djosborne** [comment]
@karya0 does `net-modules.spec` need updating?

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
-calico-node-*.tar
 # First, ignore yourself.
 .gitignore
```
Comment (ConnorDoyle) [inline .gitignore]: Not part of your changes, but why is this here?

**Comment by karya0** [comment]
I have a high-level question: why move `isolator` to `isolator/isolator`? I was imagining more like a top-level `calico` directory and calico specific bits in there without changing anything related to `isolator`.

File: .gitignore [hunk 2]
```diff
@@ -27,6 +26,10 @@ missing
 # Recommended build directory.
 /build/
 
+# Other deps
+calico-node-*.tar
+docker-compose
+
 # OSX Resource
 .DS_Store
```
Comment (ConnorDoyle) [inline .gitignore]: Again, not necessarily related to this change but this stuff IMO belongs in a user's global configuration.  No reason to have OS-specific targets here.  Mind cleaning this up?

File: Dockerfile [hunk 3]
```diff
@@ -89,3 +59,26 @@ ADD ./init_scripts/etc/service/docker/run /etc/service/docker/run
 ADD ./init_scripts/etc/service/calico/run /etc/service/calico/run
 ADD ./init_scripts/etc/service/mesos-dns/run /etc/service/mesos-dns/run
 ADD ./init_scripts/etc/config/mesos-dns.json /etc/config/mesos-dns.json
+
+
+####################
+# Isolator
+####################
+WORKDIR /isolator
+ADD ./isolator/ /isolator/
```
Comment (ConnorDoyle) [inline Dockerfile]: :+1:

**Comment by djosborne** [comment]
@karya0 A lot of the isolator specific code was scattered around the project's root directory, and moved around during the docker build. I moved it into an isolator folder so that the docker build steps could be reduced from
```
ADD ./isolator /isolator/isolator/
ADD ./m4 /isolator/m4/
ADD ./bootstrap /isolator/
ADD ./calico_isolator /isolator/
ADD ./configure.ac /isolator/
ADD ./Makefile.am /isolator/
``` 

to :
```
ADD ./isolator/ /isolator/
```

File: Makefile [hunk 4]
```diff
@@ -10,14 +10,14 @@ docker-compose:
...
 	docker pull calico/node:$(CALICO_NODE_VERSION)
-	docker save -o calico-node-$(CALICO_NODE_VERSION).tar calico/node:$(CALICO_NODE_VERSION)
+	docker save -o calico/calico-node-$(CALICO_NODE_VERSION).tar calico/node:$(CALICO_NODE_VERSION)
 
 st: images
-	for container in netmodules_marathon_1 netmodules_mesosmaster_1 netmodules_zookeeper_1 netmodules_etcd_1 netmodules_slave1_1 netmodules_slave2_1 ; do \
-		docker rm -f $$container; true; \
-	done
+	docker-compose kill
+	docker-compose -p netmodules  rm --force
```
Comment (ConnorDoyle) [inline Makefile]: nit: extra space?

File: docker-compose.yml [hunk 5]
```diff
@@ -61,11 +34,10 @@ slave2:
     - MESOS_ISOLATOR=cgroups/cpu,cgroups/mem
     - MESOS_LOG_DIR=/var/log
     - MESOS_RESOURCES=ports(*):[31000-31100]
-    - MESOS_MODULES=file:///isolator/build/isolator/modules.json
+    - MESOS_MODULES=file:///calico/modules.json
     - MESOS_ISOLATION=com_mesosphere_mesos_NetworkIsolator
     - MESOS_HOOKS=com_mesosphere_mesos_NetworkHook
     - ETCD_AUTHORITY=etcd:4001
-    - MESOS_ATTRIBUTES=public:true
```
Comment (ConnorDoyle) [inline docker-compose.yml]: I think this will break the demo, since we target the "public" slave to launch the visualizer.

File: docker-compose.yml [hunk 6]
```diff
@@ -18,36 +18,9 @@ mesosmaster:
   links:
     - zookeeper
 
-slave1:
+slave:
   build: .
   privileged: true
-  ports:
-    - "5051:5051"
```
Comment (ConnorDoyle) [inline docker-compose.yml]: Is it still possible to follow the links in the Mesos UI to the slave sandbox without this mapping?

**Comment by ConnorDoyle** [comment]
@djosborne the simplifications you made in `docker-compose.yml` are welcome, I'm just concerned they may compromise the demo.  Before I try to fire it up, were you able to verify that these changes didn't break anything?

**Comment by djosborne** [comment]
@ConnorDoyle I will confirm that the demo still works. (I was working on too many branches at once, I don't  think I reran it before I submitted this PR)

File: docker-compose.yml (hunk 5 above)
Comment (djosborne) [inline docker-compose.yml]: The demo is working if you hit `192.168.255.253:9001` directly. What is responsible for enforcing this policy on non-public slaves? Should the framework recognize that its on a non-public slave and communicate that to calico?

File: docker-compose.yml (hunk 6 above)
Comment (djosborne) [inline docker-compose.yml]: Yes, sandbox links are all still working as expected without this. Can you think of any other reasons this port mapping exists? I'm not familiar with why its here. But you can't scale a service  in docker-compose if you have a port mapping set, which is why I removed it

File: docker-compose.yml (hunk 5 above)
Comment (ConnorDoyle) [inline docker-compose.yml]: We tagged one as public in order to know deterministically which slave that task will land on since we created a route to that slave from outside the VM (via `add-container-route.sh`).  It looks like someone has since deleted that constraint from `stars.json` and `stars-isolated.json`.  Now it's up to chance that it continues to be scheduled on slave1.  I am in favor of re-introducing the attribute and the constraint to reduce the magic (relying on available resources and Marathon's current offer matching logic).

Comment (djosborne) [inline docker-compose.yml]: @ConnorDoyle 
There doesn't actually need to be a requirement that the nexthop route added via `add-container-route.sh` belongs to the Slave which the UI is running on. Since each Slave is also BGP speaker, even if the app ends up on Slave2, Slave1 will be able to nexthop the HTTP request to Slave2.

I've confirmed this by swapping the route added by `add-container-route.sh` to the other Slave and the demo still loads fine.

Comment (ConnorDoyle) [inline docker-compose.yml]: Awesome, thanks for the explanation.

File: docker-compose.yml (hunk 6 above)
Comment (ConnorDoyle) [inline docker-compose.yml]: Nope that's the only reason I can think of right now.

File: docker-compose.yml (hunk 5 above)
Comment (spikecurtis) [inline docker-compose.yml]: I think we might have used it for the "before-Calico" demo, so that we could statically port map from outside the VM.  But, we stopped showing the collector in that part of the live demo.

**Comment by spikecurtis** [comment]
LGTM!

**Comment by ConnorDoyle** [comment]
Yep, :shipit:
